gravatar-ui/src/main/java/com/gravatar/ui/components/atomic/SocialMedia.kt

```diff
@@ -0,0 +1,134 @@
+package com.gravatar.ui.components.atomic
+
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalUriHandler
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import coil.decode.SvgDecoder
+import coil.request.ImageRequest
+import com.gravatar.api.models.Account
+import com.gravatar.api.models.UserProfile
+import com.gravatar.extensions.profileUrl
+import com.gravatar.ui.R
+import java.net.MalformedURLException
+import java.net.URL
+
+public enum class LocalIcon(val shortname: String, val imageResource: Int) {
+    Gravatar("gravatar", R.drawable.gravatar_icon),
+    Calendly("calendly", R.drawable.calendly_icon),
+    Fediverse("fediverse", R.drawable.fediverse_icon),
+    Foursquare("foursquare", R.drawable.foursquare_icon),
+    Github("github", R.drawable.github_icon),
+    Instagram("instagram", R.drawable.instagram_icon),
+    Mastodon("mastodon", R.drawable.mastodongeneric_icon),
+    StackOverflow("stackoverflow", R.drawable.stackoverflow_icon),
+    TikTok("tiktok", R.drawable.tiktok_icon),
+    TripIt("tripit", R.drawable.tripit_icon),
+    Tumblr("tumblr", R.drawable.tumblr_icon),
+    Twitch("twitch", R.drawable.twitch_icon),
+    Twitter("twitter", R.drawable.twitter_icon),
+    Vimeo("vimeo", R.drawable.vimeo_icon),
+    WordPress("wordpress", R.drawable.wordpress_icon),
+    ;
+
+    companion object {
+        private val shortnames = entries.associateBy { it.shortname }
+
+        public fun valueOf(shortname: String?): LocalIcon? {
+            return shortnames[shortname]
+        }
+    }
+}
+
+public class SocialMedia(val url: URL, val name: String, val iconUrl: URL? = null, val icon: LocalIcon? = null)
```

Comment:
⛏ While reviewing, I noticed that `iconUrl` and `icon` are nullable because they could be one or the other. A solution to prevent that could be something like: 

```suggestion
public sealed class SocialMedia{
    abstract val url: URL
    abstract val name: String

    data class LocalSocialMedia(override val url: URL, override val name: String, val icon: LocalIcon) : SocialMedia()
    data class RemoteSocialMedia(override val url: URL, override val name: String, val iconUrl: URL) : SocialMedia()
}
```

Anyway, not a big deal.

Reply:
Hmmm yeah but we'll still need to watch for malformed urls and we shouldn't trust the backend anyway.

BTW, we have to be careful everytime we use the URL() constructor, maybe we can find a better way to prevent crashes.